apps/system/flash_test/flash_test.c

```diff
@@ -59,80 +59,570 @@
 #include <errno.h>
 #include <stdint.h>
 #include <sys/types.h>
-
+#include <sys/time.h>
+#include <arch/board/board.h>
+#include <tinyara/spi/spi.h>
 #include <tinyara/fs/mtd.h>
-
+#include <time.h>
+#include <tinyara/config.h>
+#include <tinyara/fs/ioctl.h>
+#include <fcntl.h>
 /****************************************************************************
  * Definitions
  ****************************************************************************/
-#define BUF_SIZE   1024
-
+#define BUF_SIZE   65536
+#define SIZE_1MB   1024*1024 
 #define READ_CMD    "read"
+#define WRITE_CMD   "write"
+#define ERASE_CMD   "erase"
+#define BCDEV_CMD	"bch"
+
+#ifdef CONFIG_BUILD_FLAT
+#define INT_FLASH_PART 7
+#define EXT_FLASH_PART 13
+#else
+#define INT_FLASH_PART 11
+#define EXT_FLASH_PART 17
+#endif
+
+
 /****************************************************************************
  * Private Data
  ****************************************************************************/
+enum flash_e {
+	FLASH_INTERNAL = 0,
+	FLASH_EXT = 1
+};
+typedef enum flash_e flash_t;
+
+FAR struct mtd_geometry_s geo;
+
+long sys_start_time = 0;
+long sys_end_time = 0;
+long systick_time_ms = 0;
+
+char blockname[32];
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+
+	gettimeofday(&tv, NULL);
```

Comment:
Let's use RTC instead of system tick. Tick is much inaccurate.

Reply:
Regarding the RTC , currently it is only seconds resolution. Also as i mentioned before , averaging may not be good idea. since RTC is in seconds resolution we might need to run the tests for minimum number of times about (1000 times) which will make us to o wait for hours :(